torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1024,15 +1033,17 @@ void LoopNest::eliminateDeadStores() {
   root_stmt_ = root_stmt_->accept_mutator(&deleter);
 }
 
-void LoopNest::prepareForCodegen() {
+void LoopNest::prepareForCodegen(
+    const c10::optional<std::unordered_set<BufPtr>>&
+        interm_bufs /*= c10::nullopt*/) {
```

Comment:
This signature is now not obvious. From external perspective it could be unclear why we need to pass a set of intermediate buffers to this function and it could be surprising that it actually controls whether we preallocate them or not. I think a cleaner interface would take a bool flag (insert alloc/free stmts or not, or in other words assume intermediate buffers are already preallocated or not).

Reply:
Is it right to assume all intermediate buffers can be preallocated, i.e., all buffers are static? Because for dynamic buffers, we can only allocate them at runtime.

Or, we can re-check the shapes of each intermediate buffer in `insertAllocFree` and always allocate buffers if dynamic; and use the bool flag to specify static buffers.